corehq/apps/app_manager/xpath_validator/wrapper.py

```diff
@@ -17,8 +17,7 @@ def validate_xpath(xpath, allow_case_hashtags=False):
             cmd = ['node', path]
         p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
                              stderr=subprocess.PIPE)
-        # Collapse whitespace. '\r' mysteriously causes the process to hang in python 3.
-        stdout, stderr = p.communicate(xpath.replace('\s+', ' ').encode('utf-8'))
+        stdout, stderr = p.communicate(xpath.replace('\r', ' ').encode('utf-8'))
```

Comment:
Can leave the comment i think.

Reply:
I'll do another PR later to replace this one.